osu.Game/Rulesets/Scoring/HitResult.cs

```diff
@@ -21,47 +24,156 @@ public enum HitResult
         /// "too far in the future). It should also define when a forced miss should be triggered (as a result of no user input in time).
         /// </remarks>
         [Description(@"Miss")]
-        Miss,
+        [Order(5)]
+        Miss = 64,
 
         [Description(@"Meh")]
+        [Order(4)]
         Meh,
 
         /// <summary>
         /// Optional judgement.
         /// </summary>
         [Description(@"OK")]
+        [Order(3)]
         Ok,
 
         [Description(@"Good")]
+        [Order(2)]
         Good,
 
         [Description(@"Great")]
+        [Order(1)]
         Great,
 
         /// <summary>
         /// Optional judgement.
         /// </summary>
         [Description(@"Perfect")]
+        [Order(0)]
         Perfect,
 
         /// <summary>
         /// Indicates small tick miss.
         /// </summary>
-        SmallTickMiss,
+        [Order(11)]
+        SmallTickMiss = 128,
 
         /// <summary>
         /// Indicates a small tick hit.
         /// </summary>
+        [Description(@"S Tick")]
+        [Order(7)]
         SmallTickHit,
 
         /// <summary>
         /// Indicates a large tick miss.
         /// </summary>
-        LargeTickMiss,
+        [Order(10)]
+        LargeTickMiss = 192,
 
         /// <summary>
         /// Indicates a large tick hit.
         /// </summary>
-        LargeTickHit
+        [Description(@"L Tick")]
+        [Order(6)]
+        LargeTickHit,
+
+        /// <summary>
+        /// Indicates a small bonus.
+        /// </summary>
+        [Description("S Bonus")]
+        [Order(9)]
+        SmallBonus = 254,
+
+        /// <summary>
+        /// Indicates a large bonus.
+        /// </summary>
+        [Description("L Bonus")]
+        [Order(8)]
+        LargeBonus = 320,
+
+        /// <summary>
+        /// Indicates a miss that should be ignored for scoring purposes.
+        /// </summary>
+        [Order(13)]
+        IgnoreMiss = 384,
```

Comment:
I'm a bit confused as to where these numbers come from.. can't find a comment anywhere explaining them which is something I think would be helpful to future visitors.

Reply:
They’re just padding in-case we add intermediate values. E.g. what if we add tiny tick as we once discussed? Or a tick that awards less than a large tick but still affects combo?

The reason is purely as safety insurance to prevent future breakage, as enums are stored as consts at IL-level.

How so you want me to comment this? An in-line non-xmldoc?

Reply:
Or, should I remove it? As discussed earlier I think we can add nested objects to achieve the same effect...

Reply:
Do we refer to them using the actual numbers anywhere in storage? Does it get serialised as a number somewhere I'm missing? If not, I think removing them would be best for now.

Reply:
No, it’s strictly for how enums are stored in DLLs.

Reply:
ah, i see..

if we're already using `[Order]` is there any reason to add padding? looking at them i can't really see a *hugely* logical order anyways. may be best to just add new ones at the end for safety... or try and get a better ruleset API versioning setup by then (we probably want this eventually, requiring rulesets to update ahead of the next release, which would avoid having to make hacks like this).

Reply:
Sure I can work with ruleset versioning, will remove.